runtime/jcl/common/criusupport.c

```diff
@@ -194,105 +235,147 @@ getNativeString(J9VMThread *currentThread, j9object_t javaString, char **nativeS
 
 
 jobject JNICALL
-Java_org_eclipse_openj9_criu_CRIUSupport_checkpointJVMImpl(JNIEnv *env,
-							jclass unused,
-							jstring checkPointDir,
-							jboolean keepRunning,
-							jboolean shellJob,
-							jboolean extUnixSupport,
-							jint logLevel,
-							jstring logFile)
+Java_org_eclipse_openj9_criu_CRIUSupport_checkpointJVMImpl(JNIEnv *env, jobject criuSupport)
 {
 	J9VMThread *currentThread = (J9VMThread*)env;
 	J9JavaVM *vm = currentThread->javaVM;
 	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
 
 	jthrowable currentExceptionLocalRef = NULL;
-	jfieldID resultType = vm->criuSupportUnsupportedOperation;
+	jfieldID resultType = vm->criuGlobals->criuSupportUnsupportedOperation;
 
 	if (vmFuncs->isCheckpointAllowed(currentThread)) {
 #if defined(LINUX)
 		j9object_t cpDir = NULL;
 		j9object_t log = NULL;
+		j9object_t wrkDir = NULL;
 		IDATA dirFD = 0;
+		IDATA workDirFD = 0;
 		char directoryBuf[STRING_BUFFER_SIZE];
 		char *directoryChars = directoryBuf;
 		char logFileBuf[STRING_BUFFER_SIZE];
 		char *logFileChars = logFileBuf;
+		char workDirBuf[STRING_BUFFER_SIZE];
+		char *workDirChars = workDirBuf;
 		IDATA systemReturnCode = 0;
+		jstring imagesDir = (*env)->GetObjectField(env, criuSupport, vm->criuGlobals->criuSupportImagesDir);
+		jboolean leaveRunning = (*env)->GetBooleanField(env, criuSupport, vm->criuGlobals->criuSupportLeaveRunning);
+		jboolean shellJob = (*env)->GetBooleanField(env, criuSupport, vm->criuGlobals->criuSupportShellJob);
+		jboolean extUnixSupport = (*env)->GetBooleanField(env, criuSupport, vm->criuGlobals->criuSupportExtUnixSupport);
+		jint logLevel = (*env)->GetIntField(env, criuSupport, vm->criuGlobals->criuSupportLogLevel);
+		jstring logFile = (*env)->GetObjectField(env, criuSupport, vm->criuGlobals->criuSupportLogFile);
+		jboolean fileLocks = (*env)->GetBooleanField(env, criuSupport, vm->criuGlobals->criuSupportFileLocks);
+		jstring workDir = (*env)->GetObjectField(env, criuSupport, vm->criuGlobals->criuSupportWorkDir);
```

Comment:
I'm not convinced this is an improvement over having multiple arguments supplied by the java code.

Reply:
Perhaps calling `checkPointJVM` with arguments makes it less maintainable when new options are added as you would need to pass in the new members at the call site.

Reply:
Yes, but that detail should not be exposed to users so that mitigates that concern.

Reply:
To clarify, I'm not proposing adding parameters to `checkPointJVM()`, but rather the implementation of that method would call `checkpointJVMImpl()` with the appropriate values rather than have the native code reach back to access fields.

Reply:
To me it feels more natural to have the args passed through the receiver object, but I guess it is a design decision. For the existing implementation every time a new option is added we would add to J9CRIUGlobals, a GetFieldID, and a Get<type>Field. With the `checkPointJVMImpl` args we would add to checkPointJVMImpl definition and call. @tajila thoughts?

Reply:
@keithc-ca We will be adding more parameters in the future. Is there a certain number of parameters where you feel passing parameters via an object is better than via the call frame? I can see us needing 10+ params

Reply:
By my count, this brings the field count to 8, but those could easily and somewhat naturally be packed into 5 parameters (3 strings, an int and 4 bits). Passing the information as parameters means crossing the JNI boundary once while fetching N fields means crossing it N+1 times. To me, more parameters only makes the bias in favour of using the call frame stronger.

Reply:
Since the purpose of this API is essentially to serialize the entire process memory, the cost of JNI transitions for field accesses are negligible here. The entire call is effectively a slow path. 

The motivation for this change is to simplify the process for adding new parameters and also improve readability when there is a large number of parameters. Its the same reason we are using the builder pattern for creating the CRIUSupport instance rather than having a large parameter list in the constructor. 

We currently have 8 params, but this number will grow. I'm not sure packing the boolean params aids simplicity or readability. My preference is to pass a single instance with all the fields, but I'll defer to your judgment here.

Reply:
If we're not concerned about the overhead, then I suggest caching all the field and method references is unnecessary.

Reply:
@EricYangIBM You can revert this and use function params instead. I don't think we will need `criuGlobals` (at least not right now), since your next PR will remove a bunch of the other field/methodIDs

Reply:
Ok, done